include/decaf.h

```diff
@@ -161,9 +167,32 @@ class Interpreter : public llvm::InstVisitor<Interpreter, ExecutionResult> {
   ExecutionResult visitPHINode(llvm::PHINode &node);
   ExecutionResult visitBranchInst(llvm::BranchInst &inst);
   ExecutionResult visitReturnInst(llvm::ReturnInst &inst);
-  ExecutionResult visitCallInst(llvm::CallInst &inst) {
-    DECAF_UNIMPLEMENTED();
-  }
+  ExecutionResult visitCallInst(llvm::CallInst &inst);
+  ExecutionResult visitSelectInst(llvm::SelectInst &inst);
+
+private:
+  ExecutionResult visitExternFunc(llvm::CallInst &inst);
+
+  /**
+   * Implements decaf_assume. Docs follow:
+   *
+   * Assert that the condition is true.
+   *
+   * In cases where the symbolic executor determines that the
+   * condition could be false, it will produce a test case with
+   * concrete inputs which reproduce the failure.
+   */
+  ExecutionResult visitAssume(llvm::CallInst &inst);
+
+  /**
+   * Implements decaf_assert. Docs follow:
+   *
+   * Assume that a condition is true.
+   *
+   * This will silently remove any executions in which the condition
+   * could evaluate to false.
+   */
+  ExecutionResult visitAssert(llvm::CallInst &inst);
```

Comment:
Why does it silently remove executions? I assume it would fail those executions and log it?

Unless it's just a way to add invariants?

Reply:
In fact it looks like the implementation does log the failure

Reply:
Whoops, it looks like I mixed up the doc comments here. They should be switched between assert and assume.